cpp/zotaut/journal_timeliness_checker.cc

```diff
@@ -47,23 +49,69 @@ bool MaxDeliveredAtSmallerThanUpdateWindow(const time_t &max_delivered_at, const
 }
 
 
+
+std::unique_ptr<KeyValueDB> CreateOrOpenKeyValueDB(const std::string &name) {
+    const std::string DB_FILENAME(UBTools::GetTuelibPath() + name + ".db");
+    if (not FileUtil::Exists(DB_FILENAME))
+        KeyValueDB::Create(DB_FILENAME);
+
+    return std::unique_ptr<KeyValueDB>(new KeyValueDB(DB_FILENAME));
+}
+
+
+time_t GetLastNotificationTime(std::unique_ptr<KeyValueDB> &notified_db,  const std::string &zeder_id) {
```

Comment:
zeder_id is not unique, this should always be a combination of zeder_id+zeder_instance, or use zeder_journals.id from the DB instead